### PR TITLE
Add Python functions for atlas generation and alignment

### LINKED ISSUE #51: Add Python script for atlas generation
Summation of steps described in [atlas generation notebook](https://github.com/KitwareMedical/HASI/blob/master/examples/TemplateGenerationIterative.ipynb).

### PR DESCRIPTION
**Pull request by tbirdso**
Changes in this PR:
- Ported atlas generation steps from Jupyter Notebook example to Python functions for streamlined use
- Added pytests to verify functions
- Removed `ninja` package dependency from Python CI

Depends on [ITKShape PyPi packaging](https://github.com/slicersalt/ITKShape/pull/20).

Closes #51.

### CODE REVIEW

File: src/hasi/test_registrars.py [hunk 1]
```diff
@@ -1,12 +1,11 @@
 #!/usr/bin/env python3
 
-# Filename:     test_meshtomeshregistration.py
+# Filename:     test_registrars.py
```
Comment (dzenanz) [inline src/hasi/test_registrars.py:3]: Same as above.

File: src/hasi/hasi/align.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/env python3
+
+# Filename: align.py
```
[the remark this answers is not in the document]
Comment (tbirdso) [reply src/hasi/hasi/align.py:3]: Sounds good, I can go back and update with the NumFOCUS header. Should this be applied to all Python files in the `hasi` module (including existing registrar class implementations)?

**Comment by tbirdso** [comment]
Note that we're seeing different behavior across platforms; [Python CI on Ubuntu](https://github.com/KitwareMedical/HASI/runs/2545644264?check_suite_focus=true) observes a higher final registration convergence metric than [on macOS](https://github.com/KitwareMedical/HASI/actions/runs/828421136) or Windows (local test).

One thought is that this could have to do with `itk::PyVectorContainer` wrapping? `KdTree` requires a `PyVectorContainer` object for reading out nearest-neighbor results. `PyVectorContainer` is wrapped for `itk.UL` or `itk.ULL` key/values depending on available `VectorContainer` wrappings by platform, previously observed different wrappings for Ubuntu/Windows tests with Github CI although no behavioral differences otherwise until now.

**Comment by tbirdso** [comment]
Added `itk-shape` dependency from PyPi and squashed fixup commits.

**Comment by tbirdso** [comment]
Failures in `build-linux-python-packages` match failures in most recent `master` branch [checks](https://github.com/KitwareMedical/HASI/runs/2545602135) and appear unrelated to this PR which deals solely with Python files (no cxx). Will open an issue to track going forward.

**Pull request merged**